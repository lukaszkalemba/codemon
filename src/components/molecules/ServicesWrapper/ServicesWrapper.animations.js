const getWrapper = animate => {
  return {
    initial: 'initial',
    animate,
  };
};

const service = {
  variants: {
    animate: {
      opacity: 1,
      y: 0,
      transition: { duration: 0.8 },
    },
    initial: { opacity: 0, y: 20 },
  },
};

export default {
  getWrapper,
  service,
};
