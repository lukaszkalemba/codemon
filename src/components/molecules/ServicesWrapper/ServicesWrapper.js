import React from 'react';
import PropTypes from 'prop-types';
import { motion } from 'framer-motion';
import Service from 'components/atoms/Service';
import animations from './ServicesWrapper.animations';

const ServicesWrapper = ({ images, animation }) => {
  const wrapperAnimation = animations.getWrapper(animation);

  return (
    <motion.div {...wrapperAnimation}>
      <Service
        img={images.design.childImageSharp}
        animation={animations.service}
      >
        Projektowanie interfejs√≥w
      </Service>
      <Service
        img={images.webapp.childImageSharp}
        animation={animations.service}
      >
        Strony i aplikacje internetowe
      </Service>
      <Service
        img={images.ecommerce.childImageSharp}
        animation={animations.service}
      >
        Sklepy internetowe
      </Service>
    </motion.div>
  );
};

ServicesWrapper.defaultProps = {
  animation: {},
};

ServicesWrapper.propTypes = {
  images: PropTypes.shape({
    design: PropTypes.shape({
      childImageSharp: PropTypes.shape({}),
    }),
    webapp: PropTypes.shape({
      childImageSharp: PropTypes.shape({}),
    }),
    ecommerce: PropTypes.shape({
      childImageSharp: PropTypes.shape({}),
    }),
  }).isRequired,
  animation: PropTypes.shape({}),
};

export default ServicesWrapper;
