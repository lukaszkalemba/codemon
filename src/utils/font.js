export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
