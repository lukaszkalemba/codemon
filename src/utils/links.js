export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
