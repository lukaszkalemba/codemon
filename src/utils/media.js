export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
