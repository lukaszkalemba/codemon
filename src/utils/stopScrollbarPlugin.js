import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
