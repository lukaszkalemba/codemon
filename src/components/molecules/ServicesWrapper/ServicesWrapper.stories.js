import React from 'react';
import ServicesWrapper from './ServicesWrapper';

export const servicesWrapper = () => (
  <ServicesWrapper animation={{}} images={{}} />
);

export default {
  component: ServicesWrapper,
  title: 'Molecules|ServicesWrapper',
};
