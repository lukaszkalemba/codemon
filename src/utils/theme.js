import color from 'utils/color';
import font from 'utils/font';
import media from 'utils/media';
import zindex from 'utils/zindex';

export default {
  color,
  font,
  media,
  zindex,
};
