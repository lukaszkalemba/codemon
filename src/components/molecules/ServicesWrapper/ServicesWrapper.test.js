import React from 'react';
import { render } from 'utils/test-utils';
import ServicesWrapper from './ServicesWrapper';

describe('<ServicesWrapper />', () => {
  it('renders correctly and matches snapshot', () => {
    const fakeImages = {
      design: {
        childImageSharp: {},
      },
      webapp: {
        childImageSharp: {},
      },
      ecommerce: {
        childImageSharp: {},
      },
    };

    const { container, getByText } = render(
      <ServicesWrapper images={fakeImages} />
    );

    const designService = getByText(/projektowanie/i);
    const websitesService = getByText(/strony/i);
    const ecommerceService = getByText(/sklepy/i);

    expect(designService).toBeInTheDocument();
    expect(websitesService).toBeInTheDocument();
    expect(ecommerceService).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
