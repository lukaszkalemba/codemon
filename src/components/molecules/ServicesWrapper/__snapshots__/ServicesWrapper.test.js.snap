// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServicesWrapper /> renders correctly and matches snapshot 1`] = `
<div>
  <article
    class="sc-AxjAm cZKMjM"
  >
    <h3
      class="sc-AxiKw iAZhDh"
      style="opacity: 0; transform: translateY(20px) translateZ(0);"
    >
      Projektowanie interfejs√≥w
    </h3>
  </article>
  <article
    class="sc-AxjAm cZKMjM"
  >
    <h3
      class="sc-AxiKw iAZhDh"
      style="opacity: 0; transform: translateY(20px) translateZ(0);"
    >
      Strony i aplikacje internetowe
    </h3>
  </article>
  <article
    class="sc-AxjAm cZKMjM"
  >
    <h3
      class="sc-AxiKw iAZhDh"
      style="opacity: 0; transform: translateY(20px) translateZ(0);"
    >
      Sklepy internetowe
    </h3>
  </article>
</div>
`;
